logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/DeleteClientUseCase.kt

```diff
@@ -6,28 +6,40 @@ import com.wire.kalium.logic.data.client.ClientRepository
 import com.wire.kalium.logic.data.client.DeleteClientParam
 import com.wire.kalium.logic.functional.fold
 import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.exceptions.isInvalidCredentials
+import com.wire.kalium.network.exceptions.isMissingAuth
 
 interface DeleteClientUseCase {
     suspend operator fun invoke(param: DeleteClientParam): DeleteClientResult
 }
 
 class DeleteClientUseCaseImpl(private val clientRepository: ClientRepository) : DeleteClientUseCase {
     override suspend operator fun invoke(param: DeleteClientParam): DeleteClientResult =
-        clientRepository.deleteClient(param).fold({ failure ->
-            if (failure is NetworkFailure.ServerMiscommunication && failure.kaliumException is KaliumException.InvalidRequestError)
-                DeleteClientResult.Failure.InvalidCredentials
-            else
-                DeleteClientResult.Failure.Generic(failure)
-        }, {
-            DeleteClientResult.Success
-        })
+        clientRepository.deleteClient(param).fold(
+            {
+                handleError(it)
+            }, {
+                DeleteClientResult.Success
+            })
+
+    private fun handleError(failure: NetworkFailure): DeleteClientResult.Failure =
+        if (failure is NetworkFailure.ServerMiscommunication && failure.kaliumException is KaliumException.InvalidRequestError)
```

Comment:
can we add logs in case we have errors , so later if the users facing any issues , the logs will be more insightful